File: Problems 1-10/Euler2.py
def fibonacci(limit):
    current, previous = 0, 1
    while current <= limit:
        yield current
        current, previous = current + previous, current

def even_fibonacci(limit):
    for fib in fibonacci(limit):
        if fib % 2 == 0:
            yield fib

total = sum(x for x in even_fibonacci(4000000))
print(total)

File: Problems 1-10/Euler2.cpp
#include <iostream>

static constexpr int sumEvenFibonacci(const int notExceeding) noexcept
{
    int total = 0;
    int current = 1, previous = 0;
    while (current <= notExceeding)
    {
        if (current % 2 == 0)
            total += current;

        const int temp = current;
        current += previous;
        previous = temp;
    }

    return total;
}

int main()
{
    constexpr int total = sumEvenFibonacci(4000000);
    std::cout << total << std::endl;

    return 0;
}
